src/Raven.Server/Documents/Handlers/Admin/AdminConfigurationHandler.cs

```diff
@@ -33,32 +33,56 @@ public Task GetSettings()
             var feature = HttpContext.Features.Get<IHttpAuthenticationFeature>() as RavenServer.AuthenticateConnection;
             var status = feature?.Status ?? RavenServer.AuthenticationStatus.ClusterAdmin;
 
-            var values = new DynamicJsonArray();
+            var settingsResult = new SettingsResult();
 
             foreach (var configurationEntryMetadata in RavenConfiguration.AllConfigurationEntries.Value)
             {
                 if (scope.HasValue && scope != configurationEntryMetadata.Scope)
                     continue;
 
-                var configurationEntryValue = new ConfigurationEntryDatabaseValue(Database.Configuration, configurationEntryMetadata, status);
-
-                values.Add(configurationEntryValue.ToJson());
+                var entry = new ConfigurationEntryDatabaseValue(Database.Configuration, configurationEntryMetadata, status);
+                settingsResult.Settings.Add(entry);
             }
 
             using (ServerStore.ContextPool.AllocateOperationContext(out TransactionOperationContext context))
             {
                 using (var writer = new BlittableJsonTextWriter(context, ResponseBodyStream()))
                 {
-                    context.Write(writer, new DynamicJsonValue
-                    {
-                        [nameof(DatabaseRecord.Settings)] = values
-                    });
+                    context.Write(writer, settingsResult.ToJson());
                 }
             }
 
             return Task.CompletedTask;
         }
 
+        [RavenAction("/databases/*/admin/configuration/settings", "PUT", AuthorizationStatus.DatabaseAdmin)]
+        public async Task PutSettings()
+        {
+            ServerStore.EnsureNotPassive();
+
+            using (ServerStore.ContextPool.AllocateOperationContext(out TransactionOperationContext context))
+            {
+                var databaseSettingsJson = context.ReadForDisk(RequestBodyStream(), Constants.DatabaseSettings.StudioId);
+                
+                Dictionary<string,string> settings = new Dictionary<string, string>();
+                var prop = new BlittableJsonReaderObject.PropertyDetails();
+                
+                for (int i = 0; i < databaseSettingsJson.Count; i++)
+                {
+                    databaseSettingsJson.GetPropertyByIndex(i, ref prop);
+                    settings.Add(prop.Name, prop.Value?.ToString() ?? null);
+                }
+
+                await UpdateDatabaseRecord(context, record =>
```

Comment:
Before doing that, do we want to dry run building the RavenConfiguration for that database to see if there are no failures?

Reply:
As discussed, this endpoint will be refactored